src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -582,20 +585,50 @@ private SmugglerProgressBase.Counts ProcessCounters(SmugglerResult result)
             using (var actions = _destination.Counters())
             using (_source.GetCounterValues(out var counters))
             {
+                var batch = new CounterBatch();
+
                 foreach (var counterDetail in counters)
```

Comment:
What if we have 10,000,000 counters?
We need to flush them every now and then (say, 10,000).
Can actually do that in parallel, like we do with docs